### PR TITLE
Overload method void uploadFile(String doi, File file); to void uploadFile(String doi, InputStream is, String filename);

### PR DESCRIPTION
**Pull request by AleixMT**
Finally here is my PR to solve #12

### CODE REVIEW

**Comment by AleixMT** [comment]
I must say that I tried all the tests before and after my modifications and I found that 3 tests fail, before and after my modifications, so I deduced that these are not my fault. I tried against (this)[https://dataverse.csuc.cat/dataverse/ICIQ] dataverse instance

File: src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java [hunk 1]
```diff
@@ -69,6 +70,15 @@ public interface DatasetOperations {
 	 */
 	void uploadFile(String doi, File file);
 
+	/**
+	 * Uploads a file using a data stream.
+	 *
+	 * @param doi Identifier of the dataset that we are attacking.
```
Comment (otter606) [inline src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java:76]: I don't think 'attacking' is a good word here
```suggestion
	 * @param doi Identifier of the dataset that we are sending the data too
```

File: src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java [hunk 2]
```diff
@@ -69,6 +70,15 @@ public interface DatasetOperations {
...
 	void uploadFile(String doi, File file);
 
+	/**
+	 * Uploads a file using a data stream.
+	 *
+	 * @param doi Identifier of the dataset that we are sending the data to.
+	 * @param file Stream of data to the contents of the file to upload.
+	 * @param filename Name of the file.
+	 */
+	void uploadFile(String doi, InputStream file, String filename);
```
Comment (otter606) [inline src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java:80]:
```suggestion
	void uploadFile(String doi, InputStream inputStream, String filename);
```

File: src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java [hunk 3]
```diff
@@ -69,6 +70,15 @@ public interface DatasetOperations {
 	 */
 	void uploadFile(String doi, File file);
 
+	/**
+	 * Uploads a file using a data stream.
+	 *
+	 * @param doi Identifier of the dataset that we are sending the data to.
+	 * @param file Stream of data to the contents of the file to upload.
+	 * @param filename Name of the file.
```
Comment (otter606) [inline src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java:78]: If the input stream doesn't come from a file, but instead from a network or whatever, then the 'file' terminology isn't quite accurate. I think 'filename' is the name you want it to appear as in Dataverse, is that correct?

File: src/main/java/com/researchspace/dataverse/http/DataverseOperationsImplV1.java [hunk 4]
```diff
@@ -224,6 +225,23 @@ public void uploadFile (String doi, File file) {
 		}
 	}
 
+	@Override
+	public void uploadFile(String doi, InputStream file, String filename) {
+		FileUploader uploader = new FileUploader();
```
Comment (otter606) [inline src/main/java/com/researchspace/dataverse/http/DataverseOperationsImplV1.java:230]: Could uploadFile(string, file) call 
```
uploadFile(doi, new FileInputStream(file), file.getName())
```
in order to reduce duplication?

**Review by otter606 — COMMENTED** [review]
Looks good, I'll try to add a basic Github action to run some tests and will get this merged. Thank you!

**Pull request merged**